pkg/sql/isql/options.go

```diff
@@ -79,13 +86,18 @@ func (tc *TxnConfig) Init(opts ...TxnOption) {
 // ExecutorConfig is the configuration used by the implementation of DB to
 // set up the Executor.
 type ExecutorConfig struct {
-	sessionData *sessiondata.SessionData
+	sessionData          *sessiondata.SessionData
+	syntheticDescriptors []catalog.Descriptor
```

Comment:
I am not very familiar with this struct but from your commit message, I found this https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/internal.go#L143-L162 existing in our code. It shows we already can synthetic descriptors in an InternalExecutor and inject synthetic descriptors to it. Are you aware of this?

Reply:
I think @fqazi is adding this field just to make it easy to pass the synthetic descriptor to the internal executor, not the other way around.

Reply:
We don't fully expose the internal executor outside of the SQL package, so it makes it hard to pass in synthetic descriptors. This just gives us a clean interface of specifying it during the creation of the internal executor.